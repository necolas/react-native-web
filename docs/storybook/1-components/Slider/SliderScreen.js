/* eslint-disable react/jsx-sort-props */

/**
 * @flow
 */
import PropDisabled from './examples/PropDisabled';
import PropMaximumTrackTintColor from './examples/PropMaximumTrackTintColor';
import PropMinimumTrackTintColor from './examples/PropMinimumTrackTintColor';
import PropOnValueChange from './examples/PropOnValueChange.js';
import PropStep from './examples/PropStep';
import PropValue from './examples/PropValue';
import React from 'react';
import UIExplorer, {
  AppText,
  Description,
  DocItem,
  Section,
  storiesOf
} from '../../ui-explorer';

const SliderScreen = () => (
  <UIExplorer title="Slider" url="1-components/Slider">
    <Description>
      <AppText>
        A component used to select a single value from a range of values.
      </AppText>
    </Description>
    <Section title="Props">
      <DocItem name="...View props" />

      <DocItem
        name="disabled"
        typeInfo="?boolean = false"
        description="If true, the user won't be able to interact with the slider"
        example={{
          render: () => <PropDisabled />
        }}
      />

      <DocItem
        name="maximumTrackTintColor"
        typeInfo="?color = #939393"
        description="The color used for the track to the right of the button."
        example={{
          render: () => <PropMaximumTrackTintColor />
        }}
      />

      <DocItem
        name="maximumValue"
        typeInfo="?number = 1"
        description="Initial maximum value of the slider."
      />

      <DocItem
        name="minimumTrackTintColor"
        typeInfo="?color = #009688"
        description="The color used for the track to the left of the button, including the button color."
        example={{
          render: () => <PropMinimumTrackTintColor />
        }}
      />

      <DocItem
        name="minimumValue"
        typeInfo="?number = 0"
        description="Initial minimum value of the slider."
      />

      <DocItem
        name="onSlidingComplete"
        typeInfo="?function"
        description="Callback that is called when the user releases the slider, regardless if the value has changed. The current value is passed as an argument to the callback handler."
      />

      <DocItem
        name="onValueChange"
        typeInfo="?function"
        description="Callback continuously called while the user is dragging the slider."
        example={{
          render: () => <PropOnValueChange />
        }}
      />

      <DocItem
        name="step"
        typeInfo="?number = 0"
        description="Step value of the slider. The value should be between 0 and (maximumValue - minimumValue)."
        example={{
          render: () => <PropStep />
        }}
      />

      <DocItem
        name="style"
        typeInfo="?ViewPropTypes.style"
        description="Used to style and layout the Slider."
      />

      <DocItem
        name="value"
        typeInfo="?number = 0"
        description="The value of the slider. Default is 0."
        example={{
          render: () => <PropValue />
        }}
      />

    </Section>
  </UIExplorer>
);

storiesOf('Components', module).add('Slider', SliderScreen);
