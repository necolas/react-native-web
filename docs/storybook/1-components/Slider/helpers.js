/**
 * @flow
 */
import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  row: {
    width: 200
  }
});
