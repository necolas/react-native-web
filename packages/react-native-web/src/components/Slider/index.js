import UnimplementedView from '../UnimplementedView';
export default UnimplementedView;
