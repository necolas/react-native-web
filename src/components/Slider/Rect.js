/**
 * @flow
 */

class Rect {
  x: number;
  y: number;
  width: number;
  height: number

  constructor(x: number, y: number, width: number, height: number) {
    this.x = x;
    this.y = y;
    this.width = width;
    this.height = height;
  }

  containsPoint(x: number, y: number) {
    return x >= this.x &&
      y >= this.y &&
      x <= this.x + this.width &&
      y <= this.y + this.height;
  }
}

export default Rect;
