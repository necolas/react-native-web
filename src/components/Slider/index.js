/**
 * Thanks to Jean Regisser <jean.regisser@gmail.com>
 * Reference: https://github.com/jeanregisser/react-native-slider
 *
 * @providesModule Slider
 * @flow
 */

import applyNativeMethods from '../../modules/applyNativeMethods';
import View from '../View';
import ViewPropTypes from '../View/ViewPropTypes';
import StyleSheet from '../../apis/StyleSheet';
import Animated from '../../apis/Animated';
import PanResponder from '../../apis/PanResponder';
import Easing from '../../apis/Easing';
import Rect from './Rect';
import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

const TRACK_SIZE = 4;
const THUMB_SIZE = 20;
const THUMB_TOUCH_SIZE = 40;

const DEFAULT_ANIMATION_CONFIGS = {
  spring: {
    friction: 7,
    tension: 100
  },
  timing : {
    duration: 150,
    easing: Easing.inOut(Easing.ease),
    delay: 0
  }
};

class Slider extends PureComponent {
  _panResponder: PanResponder;
  _previousLeft: number;
  _store: { [key: string]: { width: number, height: number, [x: string]: any } } = {};

  static propTypes = {
    /**
     * Set to true to animate values with default 'timing' animation type
     */
    animateTransitions : PropTypes.bool,

    /**
     * Used to configure the animation parameters.  These are the same parameters in the Animated library.
     */
    animationConfig : PropTypes.object,

    /**
     * Custom Animation type. 'spring' or 'timing'.
     */
    animationType : PropTypes.oneOf(['spring', 'timing']),

    /**
     * Set this to true to visually see the thumb touch rect in green.
     */
    debugTouchArea: PropTypes.bool,

    disabled: PropTypes.bool,
    maximumTrackTintColor: PropTypes.string,
    maximumValue: PropTypes.number,
    minimumTrackTintColor: PropTypes.string,
    minimumValue: PropTypes.number,
    onSlidingComplete: PropTypes.func,
    onValueChange: PropTypes.func,
    step: PropTypes.number,
    style: ViewPropTypes.style,
    testID: PropTypes.string,
    value: PropTypes.number
  };

  static defaultProps = {
    value: 0,
    minimumValue: 0,
    maximumValue: 1,
    step: 0,
    minimumTrackTintColor: '#009688',
    maximumTrackTintColor: '#939393',
    debugTouchArea: false,
    animationType: 'timing'
  };

  state = {
    containerSize: { width: 0, height: 0 },
    trackSize: { width: 0, height: 0 },
    thumbSize: { width: 0, height: 0 },
    allMeasured: false,
    value: new Animated.Value(this.props.value),
  };

  componentWillMount() {
    this._panResponder = PanResponder.create({
      onStartShouldSetPanResponder: this._handleStartShouldSetPanResponder,
      onMoveShouldSetPanResponder: this._handleMoveShouldSetPanResponder,
      onPanResponderGrant: this._handlePanResponderGrant,
      onPanResponderMove: this._handlePanResponderMove,
      onPanResponderRelease: this._handlePanResponderEnd,
      onPanResponderTerminationRequest: this._handlePanResponderRequestEnd,
      onPanResponderTerminate: this._handlePanResponderEnd,
    });
  };

  componentWillReceiveProps(nextProps) {
    const newValue = nextProps.value;

    if (this.props.value !== newValue) {
      if (this.props.animateTransitions) {
        this._setCurrentValueAnimated(newValue);
      }
      else {
        this._setCurrentValue(newValue);
      }
    }
  };

  render() {
    const {
      minimumValue,
      maximumValue,
      minimumTrackTintColor,
      maximumTrackTintColor,
      style,
      debugTouchArea,
      /* eslint-disable */
      animationType,
      /* eslint-enable */
      ...other
    } = this.props;
    const { value, containerSize, thumbSize, allMeasured } = this.state;
    const thumbLeft = value.interpolate({
      inputRange: [minimumValue, maximumValue],
      outputRange: [0, containerSize.width - thumbSize.width],
    });
    const valueVisibleStyle = {};
    if (!allMeasured) {
      valueVisibleStyle.opacity = 0;
    }

    const minimumTrackStyle = {
      position: 'absolute',
      width: Animated.add(thumbLeft, thumbSize.width / 2),
      backgroundColor: minimumTrackTintColor,
      ...valueVisibleStyle
    };

    const touchOverflowStyle = this._getTouchOverflowStyle();

    return (
      <View {...other} onLayout={this._measureContainer} style={[defaultStyles.container, style]}>
        <View
          onLayout={this._measureTrack}
          style={[{ backgroundColor: maximumTrackTintColor }, defaultStyles.track]}
        />
        <Animated.View
          style={[defaultStyles.track, minimumTrackStyle]} />
        <Animated.View
          onLayout={this._measureThumb}
          style={[
            { backgroundColor: minimumTrackTintColor },
            defaultStyles.thumb,
            {
              transform: [
                { translateX: thumbLeft },
                { translateY: 0 }
              ],
              ...valueVisibleStyle
            }
          ]}
        />
        <View
          style={[defaultStyles.touchArea, touchOverflowStyle]}
          {...this._panResponder.panHandlers}>
          {debugTouchArea === true && this._renderDebugThumbTouchRect(thumbLeft)}
        </View>
      </View>
    );
  };

  _getPropsForComponentUpdate(props) {
    /* eslint-disable */
    const {
      value,
      onValueChange,
      onSlidingStart,
      onSlidingComplete,
      style,
      ...otherProps,
    } = props;

    return otherProps;
    /* eslint-enable */
  };

  _handleStartShouldSetPanResponder = (e: Object, /*gestureState: Object*/): boolean => {
    // Should we become active when the user presses down on the thumb?
    return this._thumbHitTest(e);
  };

  _handleMoveShouldSetPanResponder(/*e: Object, gestureState: Object*/): boolean {
    // Should we become active when the user moves a touch over the thumb?
    return false;
  };

  _handlePanResponderGrant = (/*e: Object, gestureState: Object*/) => {
    this._previousLeft = this._getThumbLeft(this._getCurrentValue());
  };

  _handlePanResponderMove = (e: Object, gestureState: Object) => {
    if (this.props.disabled) {
      return;
    }

    this._setCurrentValue(this._getValue(gestureState));
    this._fireChangeEvent('onValueChange');
  };

  _handlePanResponderRequestEnd(e: Object, gestureState: Object) {
    // Should we allow another component to take over this pan?
    return false;
  };

  _handlePanResponderEnd = (e: Object, gestureState: Object) => {
    if (this.props.disabled) {
      return;
    }

    this._setCurrentValue(this._getValue(gestureState));
    this._fireChangeEvent('onSlidingComplete');
  };

  _measureContainer = (x: Object) => {
    this._handleMeasure('containerSize', x);
  };

  _measureTrack = (x: Object) => {
    this._handleMeasure('trackSize', x);
  };

  _measureThumb = (x: Object) => {
    this._handleMeasure('thumbSize', x);
  };

  _handleMeasure = (name: string, x: Object) => {
    const { width, height } = x.nativeEvent.layout;
    const size = { width: width, height: height };

    const currentSize = this._store[name];
    if (currentSize && width === currentSize.width && height === currentSize.height) {
      return;
    }
    this._store[name] = size;

    const store = this._store;
    if (store.containerSize && store.trackSize && store.thumbSize) {
      this.setState({
        containerSize: store.containerSize,
        trackSize: store.trackSize,
        thumbSize: store.thumbSize,
        allMeasured: true,
      })
    }
  };

  _getRatio = (value: number) => {
    return (value - this.props.minimumValue) / (this.props.maximumValue - this.props.minimumValue);
  };

  _getThumbLeft = (value: number) => {
    const ratio = this._getRatio(value);
    return ratio * (this.state.containerSize.width - this.state.thumbSize.width);
  };

  _getValue = (gestureState: Object) => {
    const length = this.state.containerSize.width - this.state.thumbSize.width;
    const thumbLeft = this._previousLeft + gestureState.dx;

    const ratio = thumbLeft / length;

    if (this.props.step) {
      return Math.max(this.props.minimumValue,
        Math.min(this.props.maximumValue,
          this.props.minimumValue + Math.round(ratio * (this.props.maximumValue - this.props.minimumValue) / this.props.step) * this.props.step
        )
      );
    } else {
      return Math.max(this.props.minimumValue,
        Math.min(this.props.maximumValue,
          ratio * (this.props.maximumValue - this.props.minimumValue) + this.props.minimumValue
        )
      );
    }
  };

  _getCurrentValue = () => {
    return this.state.value.__getValue();
  };

  _setCurrentValue = (value: number) => {
    this.state.value.setValue(value);
  };

  _setCurrentValueAnimated = (value: number) => {
    const animationType   = this.props.animationType;
    const animationConfig = Object.assign(
      {},
      DEFAULT_ANIMATION_CONFIGS[animationType],
      this.props.animationConfig,
      {toValue : value}
    );

    Animated[animationType](this.state.value, animationConfig).start();
  };

  _fireChangeEvent = (event) => {
    if (this.props[event]) {
      this.props[event](this._getCurrentValue());
    }
  };

  _getTouchOverflowSize = () => {
    const { state } = this;

    const size = {};
    if (state.allMeasured === true) {
      size.width = Math.max(0, THUMB_TOUCH_SIZE - state.thumbSize.width);
      size.height = Math.max(0, THUMB_TOUCH_SIZE - state.containerSize.height);
    }

    return size;
  };

  _getTouchOverflowStyle = () => {
    const {width, height} = this._getTouchOverflowSize();

    const touchOverflowStyle = {};
    if (width !== undefined && height !== undefined) {
      const verticalMargin = -height / 2;
      touchOverflowStyle.marginTop = verticalMargin;
      touchOverflowStyle.marginBottom = verticalMargin;

      const horizontalMargin = -width / 2;
      touchOverflowStyle.marginLeft = horizontalMargin;
      touchOverflowStyle.marginRight = horizontalMargin;
    }

    if (this.props.debugTouchArea === true) {
      touchOverflowStyle.backgroundColor = 'orange';
      touchOverflowStyle.opacity = 0.5;
    }

    return touchOverflowStyle;
  };

  _thumbHitTest = (e: Object) => {
    const nativeEvent = e.nativeEvent;
    const thumbTouchRect = this._getThumbTouchRect();
    return thumbTouchRect.containsPoint(nativeEvent.locationX, nativeEvent.locationY);
  };

  _getThumbTouchRect = () => {
    const { state } = this;
    const touchOverflowSize = this._getTouchOverflowSize();

    return new Rect(
      touchOverflowSize.width / 2 + this._getThumbLeft(this._getCurrentValue()) + (state.thumbSize.width - THUMB_TOUCH_SIZE) / 2,
      touchOverflowSize.height / 2 + (state.containerSize.height - THUMB_TOUCH_SIZE) / 2,
      THUMB_TOUCH_SIZE,
      THUMB_TOUCH_SIZE
    );
  };

  _renderDebugThumbTouchRect = (thumbLeft) => {
    const thumbTouchRect = this._getThumbTouchRect();
    const positionStyle = {
      left: thumbLeft,
      top: thumbTouchRect.y,
      width: thumbTouchRect.width,
      height: thumbTouchRect.height,
    };

    return (
      <Animated.View
        pointerEvents='none'
        style={[defaultStyles.debugThumbTouchArea, positionStyle]}
      />
    );
  };
}

const defaultStyles = StyleSheet.create({
  container: {
    height: 40,
    justifyContent: 'center',
  },
  track: {
    height: TRACK_SIZE,
    borderRadius: TRACK_SIZE / 2,
  },
  thumb: {
    position: 'absolute',
    width: THUMB_SIZE,
    height: THUMB_SIZE,
    borderRadius: THUMB_SIZE / 2,
  },
  touchArea: {
    position: 'absolute',
    backgroundColor: 'transparent',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
  },
  debugThumbTouchArea: {
    position: 'absolute',
    backgroundColor: 'green',
    opacity: 0.5,
  }
});

export default applyNativeMethods(Slider);
