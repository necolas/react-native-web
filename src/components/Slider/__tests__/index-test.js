/* eslint-env jasmine, jest */

import React from 'react';
import { shallow } from 'enzyme';
import Slider from '..';

describe('components/Slider', () => {
  describe('value', () => {
    test('default value', () => {
      const component = shallow(<Slider />);
      expect(component.state().value.__getValue()).toBe(0);
    });

    test('value from component props', () => {
      const component = shallow(<Slider value={0.5} />);
      expect(component.state().value.__getValue()).toBe(0.5);
    })
  });

  describe('trackTintColor', () => {
    test('maximumTrackTintColor', () => {
      const component = shallow(<Slider maximumTrackTintColor={'#F45D22'}/>);
      expect(component).toMatchSnapshot();
    });

    test('minimumTrackTintColor', () => {
      const component = shallow(<Slider maximumTrackTintColor={'#794BC4'}/>);
      expect(component).toMatchSnapshot();
    })
  });

  describe('step', () => {
    test('maximumValue: "10", step: "2", value: "4"', () => {
      const component = shallow(<Slider maximumValue={10} step={2} value={4} />);
      expect(component).toMatchSnapshot();
    })
  })
});
