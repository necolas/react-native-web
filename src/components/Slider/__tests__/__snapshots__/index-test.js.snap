// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Slider trackTintColor maximumTrackTintColor 1`] = `
<View
  onLayout={[Function]}
  step={0}
  style={
    Array [
      33,
      undefined,
    ]
  }
  value={0}
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#F45D22",
        },
        34,
      ]
    }
  />
  <AnimatedComponent
    style={
      Array [
        34,
        Object {
          "backgroundColor": "#009688",
          "opacity": 0,
          "position": "absolute",
          "width": AnimatedAddition {
            "_a": AnimatedInterpolation {
              "_children": Array [],
              "_interpolation": [Function],
              "_listeners": Object {},
              "_parent": AnimatedValue {
                "_animation": null,
                "_children": Array [],
                "_listeners": Object {},
                "_offset": 0,
                "_value": 0,
              },
            },
            "_b": AnimatedValue {
              "_animation": null,
              "_children": Array [],
              "_listeners": Object {},
              "_offset": 0,
              "_value": 0,
            },
            "_children": Array [],
            "_listeners": Object {},
          },
        },
      ]
    }
  />
  <AnimatedComponent
    onLayout={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#009688",
        },
        35,
        Object {
          "opacity": 0,
          "transform": Array [
            Object {
              "translateX": AnimatedInterpolation {
                "_children": Array [],
                "_interpolation": [Function],
                "_listeners": Object {},
                "_parent": AnimatedValue {
                  "_animation": null,
                  "_children": Array [],
                  "_listeners": Object {},
                  "_offset": 0,
                  "_value": 0,
                },
              },
            },
            Object {
              "translateY": 0,
            },
          ],
        },
      ]
    }
  />
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        36,
        Object {},
      ]
    }
  />
</View>
`;

exports[`components/Slider trackTintColor minimumTrackTintColor 1`] = `
<View
  onLayout={[Function]}
  step={0}
  style={
    Array [
      33,
      undefined,
    ]
  }
  value={0}
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#794BC4",
        },
        34,
      ]
    }
  />
  <AnimatedComponent
    style={
      Array [
        34,
        Object {
          "backgroundColor": "#009688",
          "opacity": 0,
          "position": "absolute",
          "width": AnimatedAddition {
            "_a": AnimatedInterpolation {
              "_children": Array [],
              "_interpolation": [Function],
              "_listeners": Object {},
              "_parent": AnimatedValue {
                "_animation": null,
                "_children": Array [],
                "_listeners": Object {},
                "_offset": 0,
                "_value": 0,
              },
            },
            "_b": AnimatedValue {
              "_animation": null,
              "_children": Array [],
              "_listeners": Object {},
              "_offset": 0,
              "_value": 0,
            },
            "_children": Array [],
            "_listeners": Object {},
          },
        },
      ]
    }
  />
  <AnimatedComponent
    onLayout={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#009688",
        },
        35,
        Object {
          "opacity": 0,
          "transform": Array [
            Object {
              "translateX": AnimatedInterpolation {
                "_children": Array [],
                "_interpolation": [Function],
                "_listeners": Object {},
                "_parent": AnimatedValue {
                  "_animation": null,
                  "_children": Array [],
                  "_listeners": Object {},
                  "_offset": 0,
                  "_value": 0,
                },
              },
            },
            Object {
              "translateY": 0,
            },
          ],
        },
      ]
    }
  />
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        36,
        Object {},
      ]
    }
  />
</View>
`;

exports[`components/Slider step maximumValue: "10", step: "2", value: "4" 1`] = `
<View
  onLayout={[Function]}
  step={2}
  style={
    Array [
      33,
      undefined,
    ]
  }
  value={4}
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#939393",
        },
        34,
      ]
    }
  />
  <AnimatedComponent
    style={
      Array [
        34,
        Object {
          "backgroundColor": "#009688",
          "opacity": 0,
          "position": "absolute",
          "width": AnimatedAddition {
            "_a": AnimatedInterpolation {
              "_children": Array [],
              "_interpolation": [Function],
              "_listeners": Object {},
              "_parent": AnimatedValue {
                "_animation": null,
                "_children": Array [],
                "_listeners": Object {},
                "_offset": 0,
                "_value": 4,
              },
            },
            "_b": AnimatedValue {
              "_animation": null,
              "_children": Array [],
              "_listeners": Object {},
              "_offset": 0,
              "_value": 0,
            },
            "_children": Array [],
            "_listeners": Object {},
          },
        },
      ]
    }
  />
  <AnimatedComponent
    onLayout={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#009688",
        },
        35,
        Object {
          "opacity": 0,
          "transform": Array [
            Object {
              "translateX": AnimatedInterpolation {
                "_children": Array [],
                "_interpolation": [Function],
                "_listeners": Object {},
                "_parent": AnimatedValue {
                  "_animation": null,
                  "_children": Array [],
                  "_listeners": Object {},
                  "_offset": 0,
                  "_value": 4,
                },
              },
            },
            Object {
              "translateY": 0,
            },
          ],
        },
      ]
    }
  />
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        36,
        Object {},
      ]
    }
  />
</View>
`;
